src/linearmixedmodel.jl

```diff
@@ -68,7 +68,8 @@ function LinearMixedModel(
     y, Xs = modelcols(form, tbl)
 
     y = reshape(float(y), (:, 1)) # y as a floating-point matrix
-    T = eltype(y)
+    T = promote_type(Float64, eltype(y))  # ensure that eltype of model matrices is at least Float64
```

Comment:
Should we mention this in the docs anywhere? In some areas, there is a tendency to use *half* precision to speed things up (although that matters less for x86-64 and modern ARM, it can make a difference on GPU, which we of course don't use).

Reply:
(this is my last comment, then I'm happy to merge)